### PR TITLE
Update provider autocomplete to use synonyms

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

Updating our provider search on Find to deal with synonyms which was previously imported and utilise the DFE autocomplete library. 

### Changes proposed in this pull request

- Remove the old autocomplete 
- Add styling (CSS and JS) for the new autocomplete
- Use the synonyms column on the provider table with the autocomplete

### Guidance to review

- https://find-pr-3392.london.cloudapps.digital/
- Search for a provider on Find using synonyms
- Does the search still work as we want it to?

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by avinhurry** [comment]
[Update provider autocomplete to use synonyms and abbreviations](https://trello.com/c/lydmSeh6/990-update-provider-autocomplete-to-use-synonyms-and-abbreviations)

File: app/helpers/find/provider_helper.rb [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
         value = provider.provider_name
-        option = "#{value} (#{provider.provider_code})"
-        SelectProvider.new(value, option)
-      end
-    end
+        name = "#{value} (#{provider.provider_code})"
 
-    SelectProvider = Struct.new('SelectProvider', :id, :name)
+        [name, value, data]
+      end.unshift([nil, nil, nil])
```
Comment (darokel) [inline app/helpers/find/provider_helper.rb:16]:
```suggestion
end.unshift(Array.new(3))
```

File: app/views/find/search/locations/_form.html.erb [hunk 2]
```diff
@@ -96,31 +96,20 @@
...
-                    class: "govuk-input",
-                    data: { qa: "provider-search" } %>
+                <% if provider_error? %>
+                   <p class="govuk-error-message" id="provider-error" data-qa="provider-error">
+                   <span class="govuk-visually-hidden">Error: </span><%= flash[:error].last %></p>
                 <% end %>
+                <% aria_described_by = "provider-hint" %>
+                <% aria_described_by += " provider-error" if provider_error? %>
+                <%= render Find::AutocompleteComponent.new(
+                                                form_field: form.govuk_select(
```
Comment (darokel) [inline app/views/find/search/locations/_form.html.erb:106]: Can we sort out the indentation here?

**Review by darokel — COMMENTED** [review]
Looks good. I remember we had some specs for some of the JS files. We should try to bring those over just to have some sort of parity/benchmark on the standard behaviour https://github.com/DFE-Digital/register-trainee-teachers/tree/main/app/webpacker/scripts/autocomplete.

File: app/helpers/find/provider_helper.rb (hunk 1 above)
Comment (avinhurry) [reply app/helpers/find/provider_helper.rb]: Done: https://github.com/DFE-Digital/publish-teacher-training/pull/3392/commits/a45b018f6654923a033415aecb5681875e178479

File: app/views/find/search/locations/_form.html.erb (hunk 2 above)
Comment (avinhurry) [reply app/views/find/search/locations/_form.html.erb]: Done: https://github.com/DFE-Digital/publish-teacher-training/pull/3392/commits/b4df43822c0850d27ce09727d0bbf7e2788f7f22

Does that ERB linter actually help us 😠

**Review by darokel — APPROVED** [review]
Nice work 🎉

**Review by victoriawinlow — APPROVED** [review]
Great work, thanks a lot

**Pull request merged**